### ISSUE TITLE
Weird warnings from Seqan in AhoCorasick

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#5404]
```
n function 'void seqan::allocate(T&, TValue*&, TSize, const seqan::Tag<TUsage>&) [with T = seqan::String<long unsigned int, seqan::Alloc<> >; TValue = long unsigned int; TSize = long unsigned int; TUsage = seqan::AllocateStorage_]',
    inlined from 'typename seqan::Value<seqan::String<TValue, seqan::Alloc<TSpec> > >::Type* seqan::_allocateStorage(seqan::String<TValue, seqan::Alloc<TSpec> >&, TSize) [with TValue = long unsigned int; TSpec = void; TSize = long unsigned int]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_alloc.h:373:13,
    inlined from 'typename seqan::Value<seqan::String<TValue, TSpec> >::Type* seqan::_reallocateStorage(seqan::String<TValue, TSpec>&, TSize) [with TValue = long unsigned int; TSpec = seqan::Alloc<>; TSize = long unsigned int]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_base.h:1676:45,
    inlined from 'typename seqan::Value<seqan::String<TValue, TSpec> >::Type* seqan::_reallocateStorage(seqan::String<TValue, TSpec>&, TSize, seqan::Generous) [with TValue = long unsigned int; TSpec = seqan::Alloc<>; TSize = long unsigned int]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_base.h:1735:51,
    inlined from 'void seqan::_reserveStorage(seqan::String<TValue, TSpec>&, TSize_, seqan::Tag<TUsage>) [with TValue = long unsigned int; TSpec = seqan::Alloc<>; TSize_ = long unsigned int; TExpand = seqan::TagGenerous_]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_base.h:1846:53,
    inlined from 'typename seqan::Size<seqan::String<TValue, TSpec> >::Type seqan::reserve(seqan::String<TValue, TSpec>&, TSize_, seqan::Tag<TUsage>) [with TValue = long unsigned int; TSpec = seqan::Alloc<>; TSize_ = long unsigned int; TExpand = seqan::TagGenerous_]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_base.h:1866:20,
    inlined from 'static typename seqan::Size<T>::Type seqan::_Resize_String<TExpand>::resize_(T&, typename seqan::Size<T>::Type) [with T = seqan::String<long unsigned int, seqan::Alloc<> >; TExpand = seqan::Tag<seqan::TagGenerous_>]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_base.h:1898:23,
    inlined from 'typename seqan::Size<seqan::String<TValue, TSpec> >::Type seqan::resize(seqan::String<TValue, TSpec>&, TSize, seqan::Tag<TUsage>) [with TValue = long unsigned int; TSpec = seqan::Alloc<>; TSize = long unsigned int; TExpand = seqan::TagGenerous_]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/string_base.h:1963:65,
    inlined from 'typename seqan::Size<T>::Type seqan::resize(T&, TSize) [with T = seqan::String<long unsigned int, seqan::Alloc<> >; TSize = long unsigned int]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/sequence/sequence_interface.h:1571:78,
    inlined from 'void seqan::_reportHit(TFinder&, const seqan::Pattern<TNeedle, seqan::Tag<seqan::FuzzyAC_> >&, seqan::PatternAuxData<TNeedle>&) [with TFinder = seqan::Finder<seqan::String<seqan::SimpleType<unsigned char, seqan::AAcid_>, seqan::Alloc<> > >; TNeedle = seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::AAcid_>, seqan::Alloc<> > >]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/include/OpenMS/ANALYSIS/ID/AhoCorasickAmbiguous.h:572:11,
    inlined from 'bool seqan::find(TFinder&, const seqan::Pattern<TNeedle, seqan::Tag<seqan::FuzzyAC_> >&, seqan::PatternAuxData<TNeedle>&) [with TFinder = seqan::Finder<seqan::String<seqan::SimpleType<unsigned char, seqan::AAcid_>, seqan::Alloc<> > >; TNeedle = seqan::StringSet<seqan::String<seqan::SimpleType<unsigned char, seqan::AAcid_>, seqan::Alloc<> > >]' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/include/OpenMS/ANALYSIS/ID/AhoCorasickAmbiguous.h:940:19,
    inlined from 'bool OpenMS::AhoCorasickAmbiguous::findNext(const FuzzyACPattern&)' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/include/OpenMS/ANALYSIS/ID/AhoCorasickAmbiguous.h:1041:49,
    inlined from 'void OpenMS::PeptideIndexing::addHits_(OpenMS::AhoCorasickAmbiguous&, const FuzzyACPattern&, const PeptideDB&, const OpenMS::String&, const OpenMS::String&, OpenMS::SignedSize, OpenMS::Int, OpenMS::PeptideIndexing::FoundProteinFunctor&) const' at /home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h:914:30:
/home/jenkins/ws/openms/PR/pyOMSDckr/OpenMS/src/openms/thirdparty/seqan/include/seqan/basic/allocator_interface.h:224:35: warning: argument 1 value '18446744073709551600' exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
 */  data = (TValue *) operator new(count * sizeof(TValue));
                       ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/7/bits/exception_ptr.h:40:0,
                 from /usr/include/c++/7/exception:142,
                 from /usr/include/c++/7/ios:39,
                 from pyopenms/pyopenms_2.cpp:625:
/usr/include/c++/7/new: In member function 'void OpenMS::PeptideIndexing::addHits_(OpenMS::AhoCorasickAmbiguous&, const FuzzyACPattern&, const PeptideDB&, const OpenMS::String&, const OpenMS::String&, OpenMS::SignedSize, OpenMS::Int, OpenMS::PeptideIndexing::FoundProteinFunctor&) const':
/usr/include/c++/7/new:120:7: note: in a call to allocation function 'void* operator new(std::size_t)' declared here
 void* operator new(std::size_t) _GLIBCXX_THROW (std::bad_alloc)
       ^~~~~~~~
```

@cbielow  Does this look serious to you?

**Comment by jpfeuffer** [comment]
https://github.com/OpenMS/OpenMS/blob/eff53eb5c73fb5971c190ab25bd36c2a3e673e01/src/openms/include/OpenMS/ANALYSIS/ID/AhoCorasickAmbiguous.h#L570

This can be negative and overflow!

**Comment by cbielow** [comment]
rather not. All calls to `reportHit_` are conditioned on hits not being empty.
If this is the place the warning comes from, its a false positive.

**Comment by jpfeuffer** [comment]
Related: https://gcc.gnu.org/bugzilla//show_bug.cgi?id=85783
So we have to suppress or check against PTR_MAX_DIFF

**Comment by cbielow** [comment]
resolved with #5776
